csv-validator-ui/src/main/scala/uk/gov/nationalarchives/csv/validator/ui/ScalaSwingHelpers.scala

```diff
@@ -17,12 +17,24 @@ import java.nio.file.Path
 import scala.swing.Dialog.Message
 import java.io.IOException
 import javax.swing.JTextField
+import scala.util.{Failure, Success, Try}
 
 /**
  * Some simple helpers to ease
  * the use of scala.swing
  */
 object ScalaSwingHelpers {
+  private def handleSelectedFile(dialogAction: Result.Value, fileChooser: FileChooser, result: Path => Try[String], dialogText: String): Unit =
+    dialogAction match {
+      case Result.Approve =>
+        result(fileChooser.selectedFile.toPath) match {
+          case Failure(e) =>
+            e.printStackTrace()
+            Dialog.showMessage(fileChooser, s"${e.getClass.getName}: ${e.getMessage}", s"Unable to ${dialogText.toLowerCase()} file", Message.Error)
```

Comment:
In case of error from the Open dialog, is this likely to say "Unable to OK file"